sw/device/silicon_creator/lib/drivers/flash_ctrl.h

```diff
@@ -126,17 +127,22 @@ typedef enum flash_ctrl_info_page {
  *
  * Defined here to be able to use in tests.
  */
-#define FLASH_CTRL_OTP_FIELD_SCRAMBLING \
-  (bitfield_field32_t) { .mask = UINT8_MAX, .index = CHAR_BIT * 0 }
-#define FLASH_CTRL_OTP_FIELD_ECC \
-  (bitfield_field32_t) { .mask = UINT8_MAX, .index = CHAR_BIT * 1 }
-#define FLASH_CTRL_OTP_FIELD_HE \
-  (bitfield_field32_t) { .mask = UINT8_MAX, .index = CHAR_BIT * 2 }
-
+#define FLASH_CTRL_OTP_FIELD_SCRAMBLING                            \
```

Comment:
The prior definition seems to indicate these fields were byte-aligned within the packed bitfield and the new definitions nybble-align the fields.

In the register definition of the flash controller, these are Mubi4 bools.  Are we officially changing the packing of the OTP field from Mubi8 to Mubi4?

Reply:
I don't think it really matters but I'd prefer keeping them byte-aligned (at least to be able to refer to them with just an address). That being said this is not something that needs fixing because the driver uses auto-generated constants to set the fields in the register correctly. @timothytrippel we can revert this change unless something was broken or there is another reason that I'm missing. Thanks!

Reply:
Sure I can revert. Just to confirm though, the fields in OTP are byte sized, and the fields in the flash_ctrl CSR are nybble sized, and the way this was written before was there was a direct copy from OTP to the flash_ctrl CSR, so the upper nybble just gets clipped off because of the mask applied when writing to the CSR. It works if you know to write a 4-bit mubi to the 8-bit OTP field, but since this wasn't documented anywhere, when I saw the 8-bit OTP field I assumed to use the 8-bit mubi (which also happened to work b/c the lower nybble of all mubi's are the same) but this makes it hard to decipher which mubi to fill the OTP fields with, and seemed like an error to me. Good to know its not hehe!

Reply:
fixed.